arangod/Aql/OptimizerRulesReplaceFunctions.cpp

```diff
@@ -355,6 +360,117 @@ AstNode* replaceNearOrWithin(AstNode* funAstNode, ExecutionNode* calcNode, Execu
                                  eEnumerate /* first */, eSortOrFilter /* last */,
                                  enumerateOutVariable, params.limit);
 }
+  
+/// @brief replace WITHIN_RECTANGLE
+AstNode* replaceWithinRectangle(AstNode* funAstNode, ExecutionNode* calcNode, ExecutionPlan* plan) {
+  auto* ast = plan->getAst();
+  
+  TRI_ASSERT(funAstNode->type == AstNodeType::NODE_TYPE_FCALL);
+  AstNode* fargs = funAstNode->getMember(0);
+  TRI_ASSERT(fargs->type == AstNodeType::NODE_TYPE_ARRAY);
+        
+  AstNode const* coll = fargs->getMemberUnchecked(0);
```

Comment:
This will crash if fargs has less than 5 members.
It should check with `if (fargs->numMembers() >= 5)` if the number of members is as expected, and bail out if not.